addons/product/tests/test_pricelist.py

```diff
@@ -373,3 +373,11 @@ def test_pricelist_rule_linked_to_product_variant(self):
 
         self.assertTrue(self.product_sofa_blue.pricelist_rule_ids)
         self.assertEqual(len(self.product_template_sofa.pricelist_rule_ids), 2)
+
+    def test_pricelist_item_requires_base_for_computation_of_rule_tip(self):
+        pricelist_item = Form(self.sale_pricelist_id.item_ids[0])
+        pricelist_item.compute_price = 'formula'
+        pricelist_item.base = False
+        self.assertEqual(pricelist_item.rule_tip, False)
+        with self.assertRaises(AssertionError):
+            pricelist_item.save()
```

Comment:
Sorry, but this test doesn't bring much and should be removed IMO.

First of all, the bug you fixed is not worth a test.
It's not critical at all, and triggering it needs a specific star alignment.
Adding a test for that is only adding unnecessary charge to the runbot.  It's a good practice to add tests when fixing bugs (not only for that though), but please consider what's meaningful to test or not.

Furthermore, regarding its content, it's testing:
* that you cannot save a form with an empty required field;
  This is tested by framework tests, doesn't make sense to be tested here honestly.
* that the tip is empty in some specific conditions;
   What 'matters' here (if testing this bug mattered) is to test that the bug doesn't appear, not the way the bug was solved.